packages/cli/src/commands/build/opennext-build.ts

```diff
@@ -0,0 +1,227 @@
+import { defineCloudflareConfig } from '@opennextjs/cloudflare';
+import { build } from '@opennextjs/cloudflare/cli/build/build';
+import { getNormalizedOptions } from '@opennextjs/cloudflare/cli/commands/utils';
+import { purgeCache } from '@opennextjs/cloudflare/overrides/cache-purge/index';
+import kvIncrementalCache from '@opennextjs/cloudflare/overrides/incremental-cache/kv-incremental-cache';
+import doQueue from '@opennextjs/cloudflare/overrides/queue/do-queue';
+import queueCache from '@opennextjs/cloudflare/overrides/queue/queue-cache';
+import doShardedTagCache from '@opennextjs/cloudflare/overrides/tag-cache/do-sharded-tag-cache';
+import consola from 'consola';
+import { build as esbuild, Plugin } from 'esbuild';
+import { execa } from 'execa';
+import { copyFile, cp, rm, writeFile } from 'node:fs/promises';
+import { createRequire } from 'node:module';
+import { join } from 'node:path';
+
+import { getModuleCliPath } from '../../lib/get-module-cli-path';
+import { mkTempDir } from '../../lib/mk-temp-dir';
+
+const WRANGLER_VERSION = '4.24.3';
+
+const require = createRequire(import.meta.url);
+
+const myResolverPlugin: Plugin = {
+  name: 'resolve-from-cli',
+  setup(b) {
+    b.onResolve({ filter: /^@opennextjs\/cloudflare(\/.*)?$/ }, (args) => {
+      // Resolve from the CLI package, not from the Next.js app
+      const resolved = require.resolve(args.path);
+
+      return { path: resolved };
+    });
+  },
+};
+
+const nodeBuiltinsExternal: Plugin = {
+  name: 'externalize-node-builtins',
+  setup(b) {
+    b.onResolve({ filter: /^node:/ }, () => ({ external: true }));
+  },
+};
+
+export async function opennextBuild() {
+  const [tmpdir, cleanup] = await mkTempDir('catalyst-build-');
+
+  try {
+    const openNextConfig = {
+      buildCommand: 'dotenv -e .env.local -- node ./scripts/generate.cjs && next build',
+      ...defineCloudflareConfig({
+        incrementalCache: kvIncrementalCache,
+        queue: queueCache(doQueue, {
+          regionalCacheTtlSec: 5,
+        }),
+        routePreloadingBehavior: 'withWaitUntil',
+        tagCache: doShardedTagCache({
+          baseShardSize: 12,
+          regionalCache: true,
+          regionalCacheTtlSec: 5,
+          shardReplication: {
+            numberOfSoftReplicas: 4,
+            numberOfHardReplicas: 2,
+            regionalReplication: {
+              defaultRegion: 'enam',
+            },
+          },
+        }),
+        enableCacheInterception: false,
+        cachePurge: purgeCache({ type: 'durableObject' }),
+      }),
+    };
+
+    const wranglerConfig = {
```

Comment:
Are we moving away from templates then? For `preview`, I need to create a copy of `wranger.jsonc` too and want to make sure we have one source of truth.

Reply:
I'm not opinionated on this, and I agree that my only requirement is that we have a single source of truth. If we want, I can read `wrangler.jsonc` from `fs`, and pass that to the `build` function from OpenNext... keep in mind, we also need to modify certain fields in `wrangler.jsonc`, so maybe we make a lib function to handle reading/writing/etc. the file. The file can have the config as an object, and then methods to write to `wrangler.jsonc` using `fs`, both using the same object as a source of truth?